packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1512,27 +1512,31 @@ export class AccountHandler {
     }
 
     // In other scenarios, fall back to the default behavior and let the user
-    // delete the account
-    const password = new this.Password(
-      authPW,
-      emailRecord.authSalt,
-      emailRecord.verifierVersion
-    );
+    // delete the account. If they have a password set, we verify it here. Users
+    // that don't have a password set will be able to delete their account without
+    // this step.
+    if (accountRecord.veriferSetAt > 0) {
+      const password = new this.Password(
+        authPW,
+        accountRecord.authSalt,
+        accountRecord.verifierVersion
+      );
 
-    const isMatchingPassword = await this.signinUtils.checkPassword(
-      emailRecord,
-      password,
-      request.app.clientAddress
-    );
-    if (!isMatchingPassword) {
-      throw error.incorrectPassword(emailRecord.email, emailAddress);
+      const isMatchingPassword = await this.signinUtils.checkPassword(
+        accountRecord,
+        password,
+        request.app.clientAddress
+      );
+      if (!isMatchingPassword && accountRecord.veriferSetAt > 0) {
```

Comment:
I _think_ we know `accountRecord.veriferSetAt > 0` is always `true` since we're checking that above in the parent `if` block on L1518

Reply:
Same question is it `verifier` or `verifer`?